common/faceting.js

```diff
@@ -698,86 +711,92 @@
              */
             function updateFacetColumn(scope) {
                 var defer = $q.defer();
-                
+
                 var appliedFilterToRow = function (f) {
                     return {
+                        isNotNull: f.isNotNull,
                         uniqueId: f.uniqueId,
                         displayname: f.displayname,
                         selected: true
-                    }
-                }
-                
+                    };
+                };
+
                 // console.log("updating FACET: " + scope.index);
-                
+
                 // facetColumn has changed so create the new reference
                 if (scope.facetColumn.isEntityMode) {
                     scope.reference = scope.facetColumn.sourceReference.contextualize.compactSelect;
                 } else {
                     scope.reference = scope.facetColumn.column.groupAggregate.entityCounts;
                 }
-                
+
                 // make sure to add the search term
                 if (scope.searchTerm) {
                     scope.reference = scope.reference.search(scope.searchTerm);
                 }
 
-                
+
                 var appliedLen = scope.facetModel.appliedFilters.length;
                 if (appliedLen >= tableConstants.PAGE_SIZE) {
                     scope.checkboxRows = scope.facetModel.appliedFilters.map(appliedFilterToRow);
                     defer.resolve(true);
-                } else {
-                    // read new data if neede                
-                    (function (uri) {
-                        scope.reference.read(appliedLen + tableConstants.PAGE_SIZE).then(function (page) {
-                            // if this is not the result of latest facet change
-                            if (scope.reference.uri !== uri) {
-                                defer.resolve(false);
+                }
+
+
+                // read new data if neede
+                (function (uri) {
+                    scope.reference.read(appliedLen + tableConstants.PAGE_SIZE).then(function (page) {
+                        // if this is not the result of latest facet change
+                        if (scope.reference.uri !== uri) {
```

Comment:
Shouldn't this be something like `page.reference.uri !== uri`?

The anon. function you defined passes in a uri variable which is `scope.reference.uri`.

Reply:
This is not something that I changed in this PR. But I think it's correct this way. If the facet changes then we are changing the `scope.reference`, therefore the `scope.reference.uri` is getting changed. But the `uri` is the link that we read the result with. So the `page.reference.uri` will always be equal to `uri`.

If you look at the other checks for flow-control in `table.js` and faceting.js, I'm always checking the captured `uri` is the same as `reference.uri` that is attached to `scope`/`vm` (and not the `page.reference.uri`).

Reply:
I see what you are saying. `scope.reference` is updated to a different reference which is the reason we are capturing it's value beforehand.